lexers/o/org.go

```diff
@@ -0,0 +1,80 @@
+package o
+
+import (
+	. "github.com/alecthomas/chroma" // nolint
+	"github.com/alecthomas/chroma/lexers/internal"
+)
+
+// Markdown lexer.
+var Org = internal.Register(MustNewLexer(
+	&Config{
+		Name:      "Org Mode",
+		Aliases:   []string{"org", "orgmode"},
+		Filenames: []string{"*.org"},
+		MimeTypes: []string{"text/org"}, // https://lists.gnu.org/r/emacs-orgmode/2017-09/msg00087.html
+	},
+	Rules{
+		"root": {
+			{`^#\s.*$`, Comment, nil},
+			// Headings
+			{`^(\*+)( COMMENT)( .*)$`, ByGroups(GenericHeading, NameBuiltin, Text), nil},
+			{`^(\*\*+)( COMMENT)( .*)$`, ByGroups(GenericSubheading, NameBuiltin, Text), nil},
+			{`^(\*+)( DONE)( .*)$`, ByGroups(GenericHeading, NameBuiltin, Text), nil},
+			{`^(\*\*+)( DONE)( .*)$`, ByGroups(GenericSubheading, NameBuiltin, Text), nil},
+			{`^(\*+)( TODO)( .*)$`, ByGroups(GenericHeading, Error, Text), nil},
+			{`^(\*\*+)( TODO)( .*)$`, ByGroups(GenericSubheading, Error, Text), nil},
+			{`^(\*)( .+?)( :[a-zA-Z0-9_@:]+:)()$`, ByGroups(GenericHeading, GenericStrong, GenericEmph, Text), nil}, // Level 1 heading with tags
+			{`^(\*)( .+)$`, ByGroups(GenericHeading, GenericStrong), nil}, // // Level 1 heading with NO tags
+			{`^(\*\*+)( .+?)( :[a-zA-Z0-9_@:]+:)()$`, ByGroups(GenericSubheading, Text, GenericEmph, Text), nil}, // Level 2+ heading with tags
+			{`^(\*\*+)( .+)$`, ByGroups(GenericSubheading, Text), nil}, // Level 2+ heading with NO tags
+			// Checkbox lists
+			{`^(\s*)([+-] )(\[[ X]\])( .+)$`, ByGroups(Text, Keyword, Keyword, UsingSelf("inline")), nil},
+			{`^( +)(\* )(\[[ X]\])( .+)$`, ByGroups(Text, Keyword, Keyword, UsingSelf("inline")), nil},
+			// Definition lists
+			{`^(\s*)([+-] )([^ \n]+ ::)( .+)$`, ByGroups(Text, Keyword, Keyword, UsingSelf("inline")), nil},
+			{`^( +)(\* )([^ \n]+ ::)( .+)$`, ByGroups(Text, Keyword, Keyword, UsingSelf("inline")), nil},
+			// Unordered lists
+			{`^(\s*)([+-] )(.+)$`, ByGroups(Text, Keyword, UsingSelf("inline")), nil},
+			{`^( +)(\* )(.+)$`, ByGroups(Text, Keyword, UsingSelf("inline")), nil},
+			// Ordered lists
+			{`^(\s*)([0-9]+[.)])( \[@[0-9]+\])( .+)$`, ByGroups(Text, Keyword, GenericEmph, UsingSelf("inline")), nil},
+			{`^(\s*)([0-9]+[.)])( .+)$`, ByGroups(Text, Keyword, UsingSelf("inline")), nil},
+			// Blocks
+			{`^(\s*#\+begin_src)( [^ \n]+)(.*\n)([\w\W]*?)(^\s*#\+end_src$)`,
+				UsingByGroup(
```

Comment:
This whole lexer borrows the structure from the Markdown lexer. So I retained this "UsingByGroup" part from there. While it seems to work, I don't understand what it's doing exactly. Why isn't ByGroups used like elsewhere? What is internal.Get doing? And later "2, 4" refer to those regex groups? What do they mean? The last part ("Comment, Comment, ..") makes sense as they look like the ByGroups args.

Reply:
Refer to the docs on `UsingByGroup()`, it should explain things.